stratum/hal/lib/common/yang_parse_tree.cc

```diff
@@ -199,11 +200,15 @@ ::util::Status YangParseTree::PerformActionForAllNonWildcardNodes(
     auto* leaf = subpath.elem_size() ? entry.second.FindNodeOrNull(subpath)
                                      : &entry.second;
     if (leaf == nullptr) {
-      // Should not happen!
-      ::util::Status status = MAKE_ERROR(ERR_INTERNAL)
-                              << "Found node without "
-                              << subpath.ShortDebugString() << " leaf!";
-      APPEND_STATUS_IF_ERROR(ret, status);
+      // This will happen if the subpath does not exist in the path.
+      // For example, trying to query node-id from all components
+      // but some components do not have node-id leaf.
+      if (!skip_if_not_exists) {
```

Comment:
I would be ok with always dropping the error.

Reply:
Sounds good, will clean up this one